test/tests/fips_test_suite/jtag_locked.rs

```diff
@@ -0,0 +1,69 @@
+// Licensed under the Apache-2.0 license
+
+use crate::common::fips_test_init_to_rom;
+use caliptra_builder::firmware;
+use caliptra_hw_model::{BootParams, InitParams, SecurityState};
+use caliptra_hw_model_types::DeviceLifecycle;
+
+fn check_jtag_accessible(
+    rom: &Vec<u8>,
+    debug_locked: bool,
+    device_lifecycle: DeviceLifecycle,
+    _expect_pass: bool,
+) {
+    let security_state = *SecurityState::default()
+        .set_debug_locked(debug_locked)
+        .set_device_lifecycle(device_lifecycle);
+
+    let mut _hw = fips_test_init_to_rom(Some(BootParams {
+        init_params: InitParams {
+            rom,
+            security_state,
+            ..Default::default()
+        },
+        ..Default::default()
+    }));
+
+    #[cfg(feature = "fpga_realtime")]
+    if _expect_pass {
+        assert!(
+            _hw.launch_openocd().is_ok(),
+            "Expected ok for debug_locked {debug_locked} {device_lifecycle:?}"
+        );
+    } else {
+        let result = _hw.launch_openocd();
+        assert!(
```

Comment:
I think this may be redundant with the "Ok" let in the match statement. But it's not going to break anything.